tests/automation/delete_account.spec.ts

```diff
@@ -15,147 +16,170 @@ import {
-  // Check if contact is available in contacts section
-  await clickOnElement({
-    window: restoringWindow,
-    strategy: 'data-testid',
-    selector: 'new-conversation-button',
-  });
-  await waitForElement(
-    restoringWindow,
-    'data-testid',
-    'module-conversation__user__profile-name',
-    1000,
-    userB.userName,
-  );
-  console.log('Contacts have been restored');
-  await forceCloseAllWindows(restoringWindows);
-});
+sessionTestTwoWindows(
+  'Delete account from device',
+  async ([windowA, windowB]) => {
+    let restoringWindows: Array<Page> | undefined;
+    try {
+      const [userA, userB] = await Promise.all([
+        newUser(windowA, 'Alice'),
+        newUser(windowB, 'Bob'),
+      ]);
+      // Create contact and send new message
+      await createContact(windowA, windowB, userA, userB);
+      // Delete all data from device
+      // Click on settings tab
+      await clickOnTestIdWithText(windowA, 'settings-section');
+      // Click on clear all data
+      await clickOnTestIdWithText(
+        windowA,
+        'clear-data-settings-menu-item',
+        'Clear Data',
+      );
+      // Keep 'Clear Device only' selection
+      // Confirm deletion by clicking Clear, twice
+      await clickOnMatchingText(windowA, 'Clear');
+      await clickOnMatchingText(windowA, 'Clear');
+      await waitForLoadingAnimationToFinish(windowA, 'loading-spinner');
+      // Wait for window to close and reopen
+      // await windowA.close();
+      restoringWindows = await openApp(1);
+      const [restoringWindow] = restoringWindows;
+      // Sign in with deleted account and check that nothing restores
+      await clickOnTestIdWithText(
+        restoringWindow,
+        'restore-using-recovery',
+        'Restore your account',
+      );
+      // Fill in recovery phrase
+      await typeIntoInput(
+        restoringWindow,
+        'recovery-phrase-input',
+        userA.recoveryPhrase,
+      );
+      // Enter display name
+      await typeIntoInput(
+        restoringWindow,
+        'display-name-input',
+        userA.userName,
+      );
+      // Click continue
+      await clickOnTestIdWithText(restoringWindow, 'continue-session-button');
+      await sleepFor(5000, true); // just to allow any messages from our swarm to show up
+      // Check if message from user B is restored
+      await waitForElement(
+        restoringWindow,
+        'data-testid',
+        'module-conversation__user__profile-name',
+        10000,
+        userB.userName,
+      );
+      // Check if contact is available in contacts section
+      await clickOnElement({
+        window: restoringWindow,
+        strategy: 'data-testid',
+        selector: 'new-conversation-button',
+      });
+      await waitForElement(
+        restoringWindow,
+        'data-testid',
+        'module-conversation__user__profile-name',
+        1000,
+        userB.userName,
+      );
+      console.log('Contacts have been restored');
+    } finally {
```

Comment:
because we specifically want the try{} to be allowed to throw to make the test fail.
If we were to add a catch it would be 
`catch (e) { throw e}` which is useless

Reply:
Cool